magma/circuit.py

```diff
@@ -516,8 +516,9 @@ def __init__(self, *largs, **kwargs):
         super(CircuitType, self).__init__(*largs, **kwargs)
         # Circuit instances are placed if within a definition.
         placer = _PlacerBlock.peek()
-        if placer:
-            placer.place(self)
+        if placer is None:
+            raise Exception("Can not instance a circuit outside a definition")
+        placer.place(self)
```

Comment:
Maybe we can do a try/except here since this should be the uncommon case? (Avoid the branch overhead in the usual case)

Reply:
Looks like PlacerBlock.peek() handles the None return logic, so the try except could be inside there looking for an `IndexError: list index out of range` (so we don't accidentally catch unrelated exceptions)

Reply:
Good points. See the lastest...think it's more pythonic this way